maps_dashboards/public/components/layer_config/custom_map_config/custom_map_source.tsx

```diff
@@ -0,0 +1,308 @@
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        layers: e.target.value,
+      },
+    });
+  };
+
+  const onChangeWMSVersion = (e: any) => {
+    setWMSVersion(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        version: e.target.value,
+      },
+    });
+  };
+
+  const onChangeWMSFormat = (e: any) => {
+    setWMSFormat(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        format: e.target.value,
+      },
+    });
+  };
+
+  const onChangeWMSStyles = (e: any) => {
+    setWMSStyles(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        styles: e.target.value,
+      },
+    });
+  };
+
+  const onChangeWMSCRS = (e: any) => {
+    setWMSCRS(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        crs: e.target.value,
+      },
+    });
+  };
+
+  const onChangeWMSBbox = (e: any) => {
+    setWMSBbox(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        bbox: e.target.value,
+      },
+    });
+  };
+
+  const isInvalidURL = (url: string): boolean => {
+    if (url === '') return false;
+    try {
+      new URL(url);
+      return false;
+    } catch (e) {
+      return true;
+    }
+  };
+
+  useEffect(() => {
+    setCustomMapURL(selectedLayerConfig.source.url);
+    setProtocol(selectedLayerConfig.source.protocol);
+    setCustomMapAttribution(selectedLayerConfig.source.attribution);
+    if (selectedLayerConfig.source.protocol === 'wms') {
+      setWMSLayers(selectedLayerConfig.source.layers);
+      setWMSVersion(selectedLayerConfig.source.version);
+      setWMSFormat(selectedLayerConfig.source.format);
+      setWMSStyles(selectedLayerConfig.source.styles);
+      setWMSCRS(selectedLayerConfig.source.crs);
+      setWMSBbox(selectedLayerConfig.source.bbox);
+    }
+  }, [selectedLayerConfig]);
+
+  useEffect(() => {
+    setCustomMapAttribution(selectedLayerConfig.source.attribution);
+  }, [selectedLayerConfig.source.attribution]);
+
+  useEffect(() => {
+    if (protocol === 'wms') {
+      setIsUpdateDisabled(isInvalidURL(customMapURL) || WMSLayers === '' || WMSVersion === '');
+    } else {
+      setIsUpdateDisabled(isInvalidURL(customMapURL));
+    }
+  }, [
+    WMSBbox,
+    WMSCRS,
+    WMSFormat,
+    WMSLayers,
+    WMSStyles,
+    WMSVersion,
+    customMapURL,
+    protocol,
+    setIsUpdateDisabled,
+  ]);
+
+  return (
+    <div>
+      <EuiPanel paddingSize="s">
+        <EuiForm>
+          <EuiFlexGrid columns={1}>
+            <EuiFlexItem>
+              <EuiFormLabel>Protocol</EuiFormLabel>
```

Comment:
I am thinking about calling it as "Custom type" . WDYT?

Reply:
will update